vscode-lean4/src/leanclient.ts

```diff
@@ -491,10 +492,10 @@ export class LeanClient implements Disposable {
         return this.running && range ? this.client?.protocol2CodeConverter.asRange(range) : undefined;
     }
 
-    getDiagnosticParams(uri: Uri, diagnostics: readonly Diagnostic[]) : PublishDiagnosticsParams {
+    async getDiagnosticParams(uri: Uri, diagnostics: readonly Diagnostic[]) : Promise<PublishDiagnosticsParams> {
         const params: PublishDiagnosticsParams = {
             uri: this.convertUri(uri)?.toString(),
-            diagnostics: this.client?.code2ProtocolConverter.asDiagnostics(diagnostics as Diagnostic[]) ?? []
+            diagnostics: await this.client?.code2ProtocolConverter.asDiagnostics(diagnostics as Diagnostic[]) ?? []
```

Comment:
this is probably due to the vscode version change, please revert this since Gabriel wants this to be done in a separate PR.